### ISSUE TITLE
How to add / remove classes

### ISSUE THREAD

**Issue by sebilasse** [dojo/widget-core#609]
Question: How can I add CSS classes in widgets events ?
Tried to do the Maquette way with `this.properties.classes` (true / false), did not work ...

**Comment by tomdye** [comment]
@sebilasse try setting some state in the event and use that state to determine wether or not a class should be passed to `this.classes` in the render function. 

``` typescript
render () {
   const { selected } = this.properties;
   return v('div', { classes: this.classes(selected ? css.selected : null) });
}
```

Have you joined the dojo 2 chat room on [https://gitter.im/dojo/dojo2](gitter.im/dojo/dojo2)? You may find that a more efficient way to get answers to such questions.

**Comment by sebilasse** [comment]
@tomdye 
Sorry, I do not get it. 
this.properties is `read-only` - that would mean -> I can't change `selected` <- (described in other issues) and have to rerender the widget with the new properties ?

In https://github.com/dojo/cli-build/issues/164 about `this.properties` / readonly :
"you can change the properties on a beforeRender" but this can't be changed when rendered, so
because I can't change `this.properties` I tried with a conditional for an object which is a public property of the widget `this.state` but nothing happened when it changed ...

**Comment by agubler** [comment]
@sebilasse can you clarify what you are trying to do? I am a little confused by "Tried to do the Maquette way with this.properties.classes (true / false), did not work."

**Comment by sebilasse** [comment]
@agubler @tomdye 
1. What I have so far working also in the widget is what you see in https://codepen.io/sebilasse/pen/qjxQjp?editors=1100
(an Accordion which does not need JS to work but should be progressively enhanced)

2. The Accordion has the structure
Accordion -> AccordionPanel -> AccordionTitle, AccordionMeta and AccordionContent

Now when an AccordionPanel becomes `active` also AccordionTitle must be `active` and get the `active` class (can't make it easier and change it to .active .title because all 20 semantic ui themes have this markup [content can become independently active etc.] )

Basically I need to say "each time the radio or checkbox of Panel changes, the title CSS class `active` should be the same than the checked state (which is also the active state in the Panel)"


This is the code of Accordion https://gist.github.com/sebilasse/5b587aa9462f1f714bcbd8476bacc660
This is the code of AccordionPanel
https://gist.github.com/sebilasse/5c080ffabc7d9ec4e3e3c877c18fd030

Now the AccordionTitle should have a "switchable" `active` class as described ...

**Comment by tomdye** [comment]
In the example I used above, if the `selected` state were maintained inside the widget itself then you could push this to `this._selected` and call `this.invalidate()` when you change the value to trigger a re-render. The value can then be used as before in `this.classes`.
In most cases however, the `selected` property would be pushed up to a parent component and passed in via properties as in the original example.

**Comment by sebilasse** [comment]
Well, sorry. It's not working, 
-> I'd just like to know how to remove / add a class on click. <-
I can't write a function for each class it is really much code [which was working fine in dijit, basically all semantic ui and all indieweb and all redaktor widgets].

**Comment by agubler** [comment]
@sebilasse you should be able to add a class based on a property that you pass to a child. For example, you could control this via the wrapping component(s) by letting the child components inform when they have been clicked (and their id/key etc). Then depending on the selected id, pass the a selected property of `true`  to the correct child?

I put together a small example of the type of thing I am talking about here https://github.com/agubler/adding-class, if it's not quite what you are talking about please feel free to fork and send a pull request that demonstrates the exact scenario you are referring to.

**Comment by sebilasse** [comment]
It says "Duplicate base theme class key 'title' detected, this could cause unexpected results
" (?)

**Comment by agubler** [comment]
@sebilasse sorry what does?

**Comment by sebilasse** [comment]
@agubler sorry. It is impossible to work here currently due to POTUS presence. No internet and basically no mobile but many helicopters. Will drive to a friend.

**Comment by sebilasse** [comment]
@agubler Made a fork of your example (thankyou) to demonstrate the problem, described it here : https://github.com/agubler/adding-class/issues/1 since this is basically closed ...
